test/conformance/parallel/configuration.go

```diff
@@ -0,0 +1,73 @@
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type yamlTimeDur time.Duration
+
+type MasterOffsetContinuous struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type TestCaseA struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type TestCaseB struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type Configuration struct {
+	MasterOffsetContinuousConfig MasterOffsetContinuous `yaml:"masteroffsetcontinuous"`
+	TestCaseAConfig              TestCaseA              `yaml:"testacontinuous"`
+	TestCaseBConfig              TestCaseB              `yaml:"testbcontinuous"`
+}
+
+var configuration Configuration
+var loaded bool
+
+func loadConfiguration(filename string, conf *Configuration) {
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Printf("yamlFile.Get err   #%v ", err)
+	}
+	err = yaml.Unmarshal(yamlFile, &conf)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+}
+
+func GetConfiguration() Configuration {
+	if loaded {
+		return configuration
+	}
+	loaded = true
+	path, err := os.Getwd()
+	if err != nil {
+		logrus.Error("can't retrieve path, trying to continue")
```

Comment:
The function is not trying to continue. So the message should be ,error in retrieve path . The calling method can decide if it wants to continue to exit